src/main/java/joules/Joules.java

```diff
@@ -0,0 +1,98 @@
+package joules;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+import joules.task.Deadline;
+import joules.task.Event;
+import joules.task.Task;
+import joules.task.TaskList;
+import joules.task.Todo;
+
+/**
+ * Entry point for the Joules chatbot application.
+ * <p>
+ * This class contains the {@code main} method that starts the chatbot,
+ * initializes stored tasks, and continuously processes user commands
+ * until the user exits with the {@code bye} command.
+ * </p>
+ */
+public class Joules {
+    /** Initial size of task list */
+    private static final int DEFAULT_CAPACITY = 100;
+
+    /** User interface for interacting with the chatbot */
+    private static final Ui UI = new Ui();
+
+    /** List of tasks representing the user's task history */
+    private static final TaskList tasks = new TaskList(DEFAULT_CAPACITY);
+
+    /**
+     * Starts the Joules chatbot
+     * This method processes commands such as {@code todo},
+     * {@code deadline}, {@code event}, {@code mark}, {@code unmark},
+     * {@code delete}, and {@code list}.
+     * The loop continues until the {@code bye} command is received.
+     *
+     * @param args Command-line arguments (not used).
+     */
+    public static void main(String[] args) {
+        UI.showWelcome();
+
+        // Load input
+        Store.loadTasks(tasks);
+
+        // Accept user commands
+        boolean bye = false;
+        while (!bye) {
+            try {
+                String input = UI.readInput();
+                String[] commands = input.split(" ");
+                if (input.equals("bye")) {
+                    bye = true;
+                    UI.showGoodbye();
+                } else if (input.equals("list")) {
+                    UI.listTasks(tasks);
+                } else if (commands[0].equals("find")) {
+                    String keyword = Parser.parseFind(input);
+                    UI.listMatchingTasks(keyword, tasks);
+                } else if (commands.length == 2 && Set.of("mark", "unmark", "delete").contains(commands[0])) {
+                    int taskNum = Parser.parseTaskNum(input, tasks);
+                    Task task = tasks.getTask(taskNum);
+                    if (commands[0].equals("mark")) {
+                        task.mark();
+                        Store.saveAll(tasks);
+                        UI.markTask(task);
+                    } else if (commands[0].equals("unmark")) {
+                        task.unmark();
+                        Store.saveAll(tasks);
+                        UI.unMarkTask(task);
```

Comment:
if the method name means "Unmark Task", you can possibly change it to unmarkTask instead for clarity.
```suggestion
                        UI.unmarkTask(task);
```